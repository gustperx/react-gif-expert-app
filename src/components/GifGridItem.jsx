import PropTypes from "prop-types";

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="animate__animated animate__bounceIn">
      <img className="rounded-lg object-cover h-40" src={url} alt={title} />
      <p className="text-center mt-4 text-sm font-sans font-medium text-cyan-600">
        {title}
      </p>
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
