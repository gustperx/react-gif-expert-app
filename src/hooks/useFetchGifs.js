import { useEffect, useState } from "react";
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loaging: true,
  });

  useEffect(() => {
    getGifs(category)
      .then((images) => {
        setState({
          data: images,
          loaging: false,
        });
      })
      .catch((err) => {
        console.error(err);
        setState({
          data: [],
          loaging: false,
        });
      });
  }, [category]);

  return state;
};
