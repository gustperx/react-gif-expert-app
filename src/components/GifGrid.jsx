import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";

import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const { data, loaging } = useFetchGifs(category);

  return (
    <>
      <h3 className="text-2xl text-cyan-700 font-medium mb-4 animate__animated animate__fadeInDown">
        {category}
      </h3>

      {loaging && (
        <p className="text-2xl text-red-500 my-2 text-center font-semibold animate__animated animate__fadeInLeft">
          Loading...
        </p>
      )}

      <div className="grid grid-cols-1 gap-4 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-5">
        {data.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>

      <hr className="my-8 font-semibold" />
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
