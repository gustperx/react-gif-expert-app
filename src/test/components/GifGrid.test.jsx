import { shallow } from "enzyme";
import { describe, expect, test, vi } from "vitest";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

vi.mock("../../hooks/useFetchGifs");

describe("Pruebas en el componenet <GifGrid />", () => {
  test("debe de hacer match con el snapshot", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loaging: true,
    });

    const wrapper = shallow(<GifGrid category="One Punch" />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar items cuando se cargan las imagenes useFetchGifs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://localhost/cualquier/cosa.jpg",
        title: "cualquier cosa",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loaging: false,
    });

    const wrapper = shallow(<GifGrid category="One Punch" />);

    // expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBeFalsy();
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
