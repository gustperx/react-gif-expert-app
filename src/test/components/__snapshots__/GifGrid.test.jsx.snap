// Vitest Snapshot v1

exports[`Pruebas en el componenet <GifGrid /> > debe de hacer match con el snapshot 1`] = `
<Fragment>
  <h3
    className="text-2xl text-cyan-700 font-medium mb-4 animate__animated animate__fadeInDown"
  >
    One Punch
  </h3>
  <p
    className="text-2xl text-red-500 my-2 text-center font-semibold animate__animated animate__fadeInLeft"
  >
    Loading...
  </p>
  <div
    className="grid grid-cols-1 gap-4 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-5"
  />
  <hr
    className="my-8 font-semibold"
  />
</Fragment>
`;
